### PR TITLE
Add support for Kafka 2.8.1 and 3.0.0

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

This PR:
* Adds support for Kafka 2.8.1 and 3.0.0
* Removes support for Kafka 2.7.0 and 2.7.1
* Uses Kafka 3.0.0 in the operators

It is currently using the RC1 of Kafka 3.0.0.

This has currently several unresolved issues:
* Topic Operator does not work
* Kafka Exporter does not work
* Consumers do not seem to work
* `KafkaBrokerConfigurationDiffTest` in CO does not work
* `TopicStoreTopologyProvider` in TO is using deprecated methods / classes

### Checklist

- [ ] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi/maintainers This PR should be now ready for review. It would be great to get it reviewed / merged soon so that we use a bit more in development before the next release.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java [hunk 1]
```diff
@@ -209,16 +226,21 @@ public void testLogDirsDefaultToDefault() {
 
     @Test
     public void testUnchangedLogDirs() {
-        List<ConfigEntry> ces = singletonList(new ConfigEntry("log.dirs", "null", false, true, false));
+        List<ConfigEntry> ces = singletonList(new ConfigEntry("log.dirs", "null"));
         KafkaBrokerConfigurationDiff kcd = new KafkaBrokerConfigurationDiff(Reconciliation.DUMMY_RECONCILIATION, getCurrentConfiguration(ces),
                 getDesiredConfiguration(ces), kafkaVersion, brokerId);
+
+        for (AlterConfigOp op : kcd.getConfigDiff()) {
+            System.out.println(op);
```
Comment (Frawless) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java:234]: leftover? If not, wouldn't be better to use logger?

File: topic-operator/pom.xml [hunk 2]
```diff
@@ -135,11 +135,6 @@
             <classifier>test</classifier>
             <scope>test</scope>
         </dependency>
-        <dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter-migrationsupport</artifactId>
-            <scope>test</scope>
-        </dependency>
```
Comment (Frawless) [inline topic-operator/pom.xml:142]: just out of curiosity - it was some old leftover and it wasn't used at all?

Comment (scholzj) [reply topic-operator/pom.xml:142]: It was needed for the EmbeddedKafkaCluster in Integration tests. But they improved it a bit in Kafka 3.0 so it is not needed anymore.

**Pull request merged**